lockfile_unix.go

```diff
@@ -14,17 +14,13 @@ import (
 )
 
 func getLockFile(path string, ro bool) (Locker, error) {
-	var fd int
-	var err error
-	if ro {
-		fd, err = unix.Open(path, os.O_RDONLY, 0)
-	} else {
-		fd, err = unix.Open(path, os.O_RDWR|os.O_CREATE, unix.S_IRUSR|unix.S_IWUSR)
-	}
+	// Create and open the lock file to catch potential permission errors
+	// etc. early.
+	fd, err := unix.Open(path, os.O_RDWR|os.O_CREATE, unix.S_IRUSR|unix.S_IWUSR)
 	if err != nil {
 		return nil, errors.Wrapf(err, "error opening %q", path)
 	}
-	unix.CloseOnExec(fd)
+	unix.Close(fd)
```

Comment:
This does work AFAICS, but please clearly document somewhere that `getLockFile` is only ever called once for a single value of `path` due to how `getLockfile` works; otherwise this `Close` would remove all locks held by this process.

(And we really must not have two different methods called `getLockfile` and `getLockFile`. That’s awful. Maybe in a separate PR, but soon.)

Reply:
> This does work AFAICS, but please clearly document somewhere that getLockFile is only ever called once for a single value of path due to how getLockfile works; otherwise this Close would remove all locks held by this process.

That's an issue that predates this PR but I can add a comment to `getLockFile`. Note that it's not an issue for the API `GetLockfile()` as the Lockers are held in map.

Reply:
> (And we really must not have two different methods called `getLockfile` and `getLockFile`. That’s awful. Maybe in a separate PR, but soon.)

https://github.com/containers/storage/pull/310 .  But the more of the code I read, the less I understand it…